pkg/resources/cloudconfig/configmap.go

```diff
@@ -106,6 +106,36 @@ func ConfigmapVsphereCSICreator(data configMapCreatorData) reconciling.NamedConf
 	}
 }
 
+func ConfigmapNutanixCSICreator(data configMapCreatorData) reconciling.NamedConfigMapCreatorGetter {
+	return func() (string, reconciling.ConfigMapCreator) {
+		return resources.CSICloudConfigConfigMapName, func(cm *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			if cm.Data == nil {
+				cm.Data = map[string]string{}
+			}
+
+			credentials, err := resources.GetCredentials(data)
+			if err != nil {
+				return nil, err
+			}
+
+			var pePort int32
+			if data.DC().Spec.Nutanix.PePort != nil {
+				pePort = *data.DC().Spec.Nutanix.PePort
+			} else {
+				// default nutanix prism element port
+				pePort = 9440
+			}
+
+			nutanixCsiConf := fmt.Sprintf("%s:%d:%s:%s", data.DC().Spec.Nutanix.PeEndpoint, pePort, credentials.Nutanix.PeUsername, credentials.Nutanix.PePassword)
+
+			cm.Labels = resources.BaseAppLabels(name, nil)
+			cm.Data[resources.CloudConfigConfigMapKey] = nutanixCsiConf
+
+			return cm, nil
+		}
+	}
+}
```

Comment:
Why are we adding this `ConfigMap`? It does not seem to be mounted in any of the addon resources that are being added and it contains a password and should probably be a secret because of that.

Reply:
It lives in the cluster namespace on the seed and will be mirrored into the usercluster by the user-cluster-cm. Its a pendant to the cloud-config of vsphere. Therefore it is possible to use the same logic that is used by vsphere.

Reply:
I get that, but why are we putting it in the user cluster? Where is it used?

Reply:
https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/cloudconfig/configmap.go#L79

Reply:
Not really happy about that but I see now why we need it. Let's keep it if that is necessary.